pkgs/stdenv/generic/setup.sh

```diff
@@ -31,7 +31,7 @@ runHook() {
     local hook
     # Hack around old bash being bad and thinking empty arrays are
     # undefined.
-    for hook in "_callImplicitHook 0 $hookName" ${!hooksSlice+"${!hooksSlice}"}; do
+    for hook in "_callImplicitHook 0 $hookName" ${!hooksSlice}; do
```

Comment:
If you were removing a hack for old bash, you’d also want to remove the comment saying there’s a hack for old bash.

However, that’s not what’s actually happening. The comment is wrong.

There is no array here. `${!hooksSlice}` is a (scalar) [indirect expansion](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html)—it expands to the value of the variable whose name is the value of `$hooksSlice`.

`${!hooksSlice+"${!hooksSlice}"}` is the correct way to quote this construction so that the `for` loop sees it exactly once if it does exist (we don’t want to split on whitespace), and skips it if it doesn’t exist (we don’t want to iterate over the empty string).